src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.HighPriorityProcessor.cs

```diff
@@ -37,29 +39,49 @@ private sealed class HighPriorityProcessor : IdleProcessor
                     {
                         _processor = processor;
                         _lazyAnalyzers = lazyAnalyzers;
+                        _gate = new object();
 
                         _running = Task.CompletedTask;
                         _workItemQueue = new AsyncDocumentWorkItemQueue(processor._registration.ProgressReporter, processor._registration.Workspace);
 
                         Start();
                     }
 
+                    public ImmutableArray<IIncrementalAnalyzer> Analyzers
+                    {
+                        get
+                        {
+                            lock (_gate)
```

Comment:
Is the lock necessary for the read? i.e. are we protecting read from write, or just concurrent reads (which can trigger initialization)?

Reply:
I copied most of this from existing code in AbstractPriorityProcessor. I will keep it the same way and be defensive and consistent.